python/seldon_core/seldon_client.py

```diff
@@ -211,10 +211,15 @@ def __init__(
         logger.debug("Configuration:" + str(self.config))
 
     def _gather_args(self, **kwargs):
-
-        c2 = {**self.config}
-        c2.update({k: v for k, v in kwargs.items() if v is not None})
-        return c2
+        """
+        Performs a left outer join where kwargs is left and self.config is right
```

Comment:
Nice way to describe it!

Reply:
Yeah the previous one was actually doing a full join, which is not really what I think we wanted, but now it does a left outer join